tests/test_utils.py

```diff
@@ -0,0 +1,56 @@
+"""Tests for avro2py/utils.py"""
+import datetime
+import uuid
+
+from hypothesis import strategies as st, given
+
+from avro2py.utils import to_avro_dict
+from example.marketprice.messages.wallyworld.bidder import (
+    WallyworldAdGroupStructure, WallyworldAdItem
+)
+
+
+@given(st.integers(), st.text(), st.decimals())
+def test_to_avro_dict_nested_named_tuples(random_int, item_id, bid):
+    requested_at = datetime.datetime.utcnow()
+    campaign_id = ad_group_id = advertiser_id = random_int
+    users_lined = [uuid.uuid4() for _ in range(3)]
```

Comment:
I believe that if this ever fails, the UUID and datetime generation within the test will frustrate Hypothesis' ability to reproduce the failure. Instead:
```suggestion
@given(
    random_int=st.integers(), 
    item_id=st.text(), 
    bid=st.decimals(),
    requested_at=st.datetimes(),
    users_lined=st.lists(st.uuid(), min_size=3, max_size=3)
)
def test_to_avro_dict_nested_named_tuples(random_int, item_id, bid, requested_at, users_lined):
    campaign_id = ad_group_id = advertiser_id = random_int
```